### DISCUSSION TITLE
Identify multiple namespaces/components as internal

### DISCUSSION THREAD

**Discussion by iamrahul127** [DependencyTrack/dependency-track#1720] [Q&A]
We would like to mark our own nuget & npm packages as internal. Managed to identify nuget packages using `'.*education.*` regex but not able to do it for npm packages which has `@education` as group. 

Regex I tried is => `.*education.*','.*@education.*'`

What am I missing here?

**Comment by iamrahul127** [comment]
Can someone please help?

**Comment by nscuro** [comment]
You need **one** regex that matches all your cases, you cannot provide lists of regexes. 

Use sites like https://regex101.com/ to test your regex (make sure to select *Java 8* in the *Flavor* section to the left).

In fact, `.*education.*` should already cover all your cases:

![grafik](https://user-images.githubusercontent.com/5693141/184107413-c75cb07a-3370-4683-b057-2ef875771499.png)

**Reply by iamrahul127** [reply]
Internal component identification for Dotnet binaries works with 'component name regex as shown in below screenshot.
Challenge is, internal angular package identification is neither working with 'component name regex' nor with 'component namespace regex'. I hope problem is more clear now.


![image](https://user-images.githubusercontent.com/24877322/184111137-ab0650eb-cde9-4921-aedc-ecd2c8cb8cf3.png)

**Reply by nscuro** [reply]
It isn't matching because your namespace regex doesn't work as you think it does. It will match `'education','@education'`, but not `education` or `@education`:

![grafik](https://user-images.githubusercontent.com/5693141/184134991-d78332aa-a3ca-4c52-962c-ab0245983834.png)

Please use the website I linked to test your regexes. Try just `.*education.*`.

**Reply by iamrahul127** [reply]
Tried what you have suggested. As you mentioned, regex `.*education.*` should identify `@education ` as internal component as that correct regex. 
But it's not identifying. My question is, does "Component namespace regex" is executed against package "Group"? Or it is only checking package name or namespace? 
I am not sure how it works with npm packages.
